modules/foreman_provisioning_template.py

```diff
@@ -0,0 +1,388 @@
+        description:
+        - The organizations the template shall be assigned to
+        required: false
+    operatingsystems:
+        description: The Operatingsystems the template shall be assigned to
+        required: false
+    state:
+        description: The state the template should be in.
+        require: true
+        choices:
+        - absent
+        - latest
+        - present
+
+'''
+
+EXAMPLES = '''
+
+# Keep in mind, that in this case, the inline parameters will be overwritten
+- name: "Create a Provisioning Template inline"
+  local_action:
+      module: foreman_provisioning_template
+      username: "admin"
+      password: "admin"
+      server_url: "https://foreman.example.com"
+      name: A New Finish Template
+      kind: finish
+      state: present
+      template: |
+        <%#
+            name: Finish timetravel
+            kind: finish
+        %>
+        cd /
+        rm -rf *
+      locations:
+      - Gallifrey
+      organizations:
+      - TARDIS INC
+
+- name: "Create a Provisioning Template from a file"
+  local_action:
+      module: foreman_provisioning_template
+      username: "admin"
+      password: "admin"
+      server_url: "https://foreman.example.com"
+      file_name: timeywimey_template.erb
+      state: present
+      locations:
+      - Gallifrey
+      organizations:
+      - TARDIS INC
+
+# Due to the module logic, deleting requires a template dummy,
+# either inline or from a file.
+- name: "Delete a Provisioning Template"
+  local_action:
+      module: foreman_provisioning_template
+      username: "admin"
+      password: "admin"
+      server_url: "https://foreman.example.com"
+      name: timeywimey_template
+      template: |
+        <%#
+            dummy:
+        %>
+      state: absent
+
+- name: "Create a Provisioning Template from a file and modify with parameter"
+  local_action:
+      module: foreman_provisioning_template
+      username: "admin"
+      password: "admin"
+      server_url: "https://foreman.example.com"
+      file_name: timeywimey_template.erb
+      name: Wibbly Wobbly Template
+      state: present
+      locations:
+      - Gallifrey
+      organizations:
+      - TARDIS INC
+
+# Providing a name in this case wouldn't be very sensible.
+# Alternatively make use of with_filetree to parse recursively with filter.
+- name: "Parsing a directory of provisioning templates"
+  local_action:
+      module: foreman_provisioning_template
+      username: "admin"
+      password: "admin"
+      server_url: "https://foreman.example.com"
+      file_name: "{{ item }}"
+      state: present
+      locations:
+      - SKARO
+      organizations:
+      - DALEK INC
+      with_fileglob:
+       - "./arsenal_templates/*.erb"
+
+'''
+
+RETURN = ''' # '''
+
+
+try:
+    from nailgun.entities import (
+        ProvisioningTemplate,
+        OperatingSystem,
+        _OPERATING_SYSTEMS,
+    )
+
+    from ansible.module_utils.ansible_nailgun_cement import (
```

Comment:
@akofink that exactly is our current workaround, but since the `ansible.cgf` also needs to be relative to the playbook rather than the role, we think this can only be solved satisfyingly within ansible core.